lib/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -640,6 +640,18 @@ protected function prepare_links( $comment ) {
 			);
 		}
 
+		// Only grab one comment to verify the comment has children.
+		$comment_children = $comment->get_children( array( 'number' => 1 ) );
```

Comment:
I tested this on 1,100,000 comments, using Redis it dropped the response time back to normal.  The query is object cached, so to my understanding it should not be a big problem.  There is very little difference in response time for that magnitude of comments when the children are grabbed or not grabbed. On average over 100 requests each the difference was 2ms.